aten/src/ATen/cpu/vec/vec256/vec256_convert.h

```diff
@@ -128,30 +128,102 @@ struct VecConvert<int32_t, 1, uint8_t, 1> {
 
 template <typename dst_t>
 struct VecConvert<
-  dst_t,
-  1,
-  float,
-  1,
-  typename std::enable_if_t<
-    std::is_same_v<dst_t, unsigned char> || std::is_same_v<dst_t, signed char>,
-    void>> {
+    dst_t,
+    1,
+    BFloat16,
+    1,
+    typename std::enable_if_t<
+        std::is_same_v<dst_t, unsigned char> ||
+            std::is_same_v<dst_t, signed char>,
+        void>> {
   static inline VectorizedN<dst_t, 1> apply(
-      const VectorizedN<float, 1>& src) {
-    return convert_float_to_int8<dst_t>(src[0]);
+      const VectorizedN<BFloat16, 1>& src) {
+    __m256 value;
+    cvtbf16_fp32(_mm256_castsi256_si128(src[0]), value);
+    return convert_float_to_int8<dst_t>(value);
```

Comment:
why not calling the existing convert from int8 to fp32 and fp32 to bf16/fp16? Then you can combine bf16 and fp16 into a single implementation.

Reply:
Thanks for your advice. Modified it!